### PR TITLE
Add filter argument to monitor script

### PR DESCRIPTION
**Pull request by merwan**
- replace flake8 by ruff
- Improve CI to run for every supported Python version
- Explicitly export public names as required by pyright or mypy: https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface
- Add `filter_fields` argument to `monitor.py` script to filter events for a specific field value

### CODE REVIEW

File: scripts/monitor.py [hunk 1]
```diff
@@ -51,6 +52,12 @@ def dump(self, key, conversation_id, payload):
             print("Payload:")
             pprint.pprint(payload)
 
+    def _should_skip(self, payload):
+        for field, value in self.filter_fields:
+            if payload.get(field) != value:
+                return True
+        return False
```
Comment (rtxm) [inline scripts/monitor.py:59]: What about this:

```suggestion
    def _should_skip(self, payload):
        return any(payload.get(field) != value for field, value in self.filter_fields)
```

File: src/eventail/async_service/pika/base.py [hunk 2]
```diff
@@ -605,7 +605,7 @@ def on_message_returned(
         """
         decoder = cbor if properties.content_type == "application/cbor" else json
         # If we are not able to decode our own payload, better crash the service now
-        payload: JSON_MODEL = decoder.loads(body) if body else None
+        payload: JSON_MODEL = decoder.loads(body) if body else None  # type: ignore
```
Comment (rtxm) [inline src/eventail/async_service/pika/base.py:608]: Is mypy OK with this now?

**Review by rtxm — APPROVED** [review]
Thanks for your contribution!
Looks good to me. I made a suggestion though.

File: scripts/monitor.py (hunk 1 above)
Comment (merwan) [reply scripts/monitor.py:59]: Thanks for the suggestion, I was not happy with the previous code, this is cleaner!

**Pull request merged**